src/opts/mod.rs

```diff
@@ -68,6 +68,15 @@ pub const DEFAULT_TTL_CHECK_INTERVAL: Duration = Duration::from_secs(30);
 pub(crate) enum HostPortOrUrl {
     HostPort(String, u16),
     Url(Url),
+    /// Specify an already-resolved host and port, useful for ensuring that the specified IP
+    /// addresses are used for the TCP connection.
+    ResolvedHost {
+        /// The hostname to use for SSL certificate verification.
+        hostname: String,
+        /// The resolved IP addresses to use for the TCP connection.
+        ips: Vec<IpAddr>,
+        port: u16,
+    },
```

Comment:
Does this need to be a separate variant? It seems like it might be cleaner to change `HostPort(String, u16)` to:

```
HostPort {
    hostname: String,
    port: u16,
    /// The resolved IP addresses to use for the TCP connection. If empty,
    /// DNS resolution of `hostname` will be performed.
    resolved_ips: Vec<IpAddr>,
}
```